cmd/oracled/cmd/register_oracle.go

```diff
@@ -43,18 +42,20 @@ func registerOracleCmd() *cobra.Command {
 				return err
 			}
 
-			queryClient, err := panacea.NewVerifiedQueryClient(context.Background(), conf, trustedBlockInfo)
+			sgx := sgx.NewOracleSgx()
+
+			queryClient, err := panacea.NewVerifiedQueryClient(context.Background(), conf, trustedBlockInfo, sgx)
 			if err != nil {
 				return fmt.Errorf("failed to create queryClient: %w", err)
 			}
 
-			svc, err := service.NewWithQueryClient(conf, queryClient)
+			svc, err := service.New(conf, sgx, queryClient)
 			if err != nil {
 				return fmt.Errorf("failed to create service: %w", err)
 			}
 			defer svc.Close()
```

Comment:
This practice is not recommended. If you create/open SGX and QueryClient outside of `service.New()`, those resources should be closed explicitly outside of `svc.Close()`. Previously, everything has been closed by `svc.Close()` since all of them have been initialized in `service.New()`.
In other words, resources should be closed in the layer where they were created.
If we don't make this practice clearly, we would meet some issues later, regarding accessing closed resources or double-closing resources.

Reply:
Yes. My intention was to externally control the QueryClient or Sgx.
It was for reuse.

Right now the service is a huge module and we use it all over the place, so I think it's better to create it internally as you said.
But if I need to reuse it in the future, it seems like a good idea to separate it.

Reply:
I mean that it's okay to manage resources separately (outside of Service). But, if so, we should explicitly close those resources separately with the Service. In your code, the QueryClient is closed in the `svc.Close`.

Reply:
OK. I miss understood it. 
As you suggested, I will call the Close function outside of the service.